### PR TITLE
Replace `nb_available` with `bytesavailable`

### PR DESCRIPTION
**Pull request by ReubenHill**
`nb_available` has been deprecated in 1.0/0.7. This was missed in previous 0.7/1.0 compatibility PRs due to a depwarning not being thrown. `bytesavailable` is now used instead.

### CODE REVIEW

File: src/LibFTD2XX.jl [hunk 1]
```diff
@@ -138,7 +138,7 @@ function Base.close(handle::FT_HANDLE)
 end
 
 function Base.readbytes!(handle::FT_HANDLE, b::AbstractVector{UInt8}, nb=length(b))
-  nbav = nb_available(handle)
+  @compat nbav = bytesavailable(handle)
```
Comment (samuelpowell) [inline src/LibFTD2XX.jl]: `@compat` not required

File: src/LibFTD2XX.jl [hunk 2]
```diff
@@ -200,19 +200,30 @@ function status(handle::FT_HANDLE)
...
 end
 
-function Base.nb_available(handle::FT_HANDLE)
-  nbrx = Ref{DWORD}()
-  status = ccall(cfunc[:FT_GetQueueStatus], cdecl, FT_STATUS, 
-                 (FT_HANDLE, Ref{DWORD}),
-                  handle,    nbrx)
-  FT_STATUS_ENUM(status) == FT_OK || throw(FT_STATUS_ENUM(status))
-  nbrx[]
+if VERSION < v"0.7"
```
Comment (samuelpowell) [inline src/LibFTD2XX.jl]: Is this required? Surely `Compat` will map this name to `bytesavailable`, including the new method?

File: src/LibFTD2XX.jl [hunk 3]
```diff
@@ -200,19 +200,30 @@ function status(handle::FT_HANDLE)
...
+    status = ccall(cfunc[:FT_GetQueueStatus], cdecl, FT_STATUS, 
+                   (FT_HANDLE, Ref{DWORD}),
+                    handle,    nbrx)
+    FT_STATUS_ENUM(status) == FT_OK || throw(FT_STATUS_ENUM(status))
+    nbrx[]
+  end
 end
 
-Base.eof(handle::FT_HANDLE) = (nb_available(handle) == 0)
+@compat Base.eof(handle::FT_HANDLE) = (bytesavailable(handle) == 0)
```
Comment (samuelpowell) [inline src/LibFTD2XX.jl]: `@compat` not required

File: src/LibFTD2XX.jl [hunk 4]
```diff
@@ -200,19 +200,30 @@ function status(handle::FT_HANDLE)
...
 
 function Base.readavailable(handle::FT_HANDLE)
-  @compat b = Vector{UInt8}(undef, nb_available(handle))
+  @compat b = Vector{UInt8}(undef, bytesavailable(handle))
```
Comment (samuelpowell) [inline src/LibFTD2XX.jl]: Please write 

```b = @compat Vector{UInt8}(undef, bytesavailable(handle))```

Just so we know exactly what the transform is

**Review by samuelpowell — CHANGES_REQUESTED** [review]

File: src/LibFTD2XX.jl (hunk 2 above)
Comment (ReubenHill) [reply src/LibFTD2XX.jl]: `Base.bytesavailable` was not in name space on v0.6, needed `Compat.bytesavailable` to work.

File: src/LibFTD2XX.jl (hunk 4 above)
Comment (ReubenHill) [reply src/LibFTD2XX.jl]: done

**Review by samuelpowell — APPROVED** [review]

**Pull request merged**